### ISSUE TITLE
Question: can I only display first line of prompt when it changes?

### ISSUE THREAD

**Issue by TamasBarta** [Powerlevel9k/powerlevel9k#1357]
I want to achieve a cleaner terminal output, and I thought I want to try printing the first line of the prompt only when it changes, so I still have the info I need, but it isn't printed redundantly over and over with every command.

Is it currently achievable? Thanks!

**Comment by romkatv** [comment]
Suppose your prompt looks like this:

```
~/powerlevel9k master >
```

You type `ls` and hit `ENTER`. The output of `ls` appears on the screen. How should the next prompt look like? Empty line? Just `>`? Something else?

**Comment by TamasBarta** [comment]
I have multiple lines in my prompt. One line for the segments, and one for just a `$` that is an actual prompt.

It looks rather like this:

```
 ~/Development/project > my-branch >
 $ 
```

I only want the first line to appear only on change. The `$` should be persistent, so I know when a command finished executing.

**Comment by romkatv** [comment]
So, after `ls` prints its output you want only `$` as prompt?

```
~/Development/project > my-branch >
$ ls
foo bar baz

$ █
```

**Comment by TamasBarta** [comment]
Exactly, only without the empty line as well if possible. Deep in folders my terminal is crowded by the colourful backgrounds of segments, it may fix that. Sorry if I wasn't clear enough.

**Comment by TamasBarta** [comment]
Thanks for the tips, @romkatv! I'll try these, but I also hope the original idea gets some development effort.

**Comment by romkatv** [comment]
> I also hope the original idea gets some development effort.

The last commit to Powerlevel9k was made 5 months ago. Set your expectations accordingly.

**Comment by FernandoGarcia** [comment]
Hi!
@romkatv I'm interested in this feature too. Can you add it to your 10k theme?
Best regards.

**Comment by romkatv** [comment]
> @romkatv I'm interested in this feature too. Can you add it to your 10k theme?

I don't particularly like the idea of showing prompt only when it changes but I can see the value in using different formats for the current and the past prompts. The latter is something I could add. The configuration might look like this:

```zsh
# Current left prompt.
POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(dir vcs prompt_char)

# Past left prompt.
POWERLEVEL9K_PAST_LEFT_PROMPT_ELEMENTS=(prompt_char)
```

Keep in mind that you can alleviate the problem of heavy-weight past prompts by running `p10k configure` and choosing `lean` style with few icons. This is my personal favorite.

**Comment by FernandoGarcia** [comment]
Thanks for your answer.
